jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1455,9 +1455,9 @@ public void succeeded()
                 // We are being tough on handler implementations and expect them
                 // to not have pending operations when calling succeeded or failed.
                 if (httpChannelState._onContentAvailable != null)
-                    throw new IllegalStateException("demand pending");
+                    failure = ExceptionUtil.combine(failure, new IllegalStateException("demand pending"));
```

Comment:
You need to change the comment above.  We are now being less tough as we are converting a call to `succeeded` with pending demand/write into a call to `failure`.

Reply:
Right, I've modified the comment accordingly.